src/main/java/ch/njol/skript/effects/EffSendTitle.java

```diff
@@ -32,90 +34,84 @@
-		else
-			Skript.registerEffect(EffSendTitle.class,
-					"send title %string% [with subtitle %-string%] [to %players%]",
-					"send subtitle %string% [to %players%]");
+		Skript.registerEffect(EffSendTitle.class,
+			"send title %object% [with subtitle %-object%] [to %players%] [for %-timespan%] [with fade[(-| )]in %-timespan%] [[and] [with] fade[(-| )]out %-timespan%]",
+			"send subtitle %object% [to %players%] [for %-timespan%] [with fade[(-| )]in %-timespan%] [[and] [with] fade[(-| )]out %-timespan%]");
 	}
-	
+
 	@Nullable
-	private Expression<String> title;
+	private Expression<?> title;
 	@Nullable
-	private Expression<String> subtitle;
+	private Expression<?> subtitle;
 	@SuppressWarnings("null")
 	private Expression<Player> recipients;
 	@Nullable
 	private Expression<Timespan> fadeIn, stay, fadeOut;
-	
+
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		title = matchedPattern == 0 ? (Expression<String>) exprs[0] : null;
-		subtitle = (Expression<String>) exprs[1 - matchedPattern];
+		if (matchedPattern == 0) title = LiteralUtils.defendExpression(exprs[0]);
+		subtitle = LiteralUtils.defendExpression(exprs[1 - matchedPattern]);
 		recipients = (Expression<Player>) exprs[2 - matchedPattern];
-		if (TIME_SUPPORTED) {
-			stay = (Expression<Timespan>) exprs[3 - matchedPattern];
-			fadeIn = (Expression<Timespan>) exprs[4 - matchedPattern];
-			fadeOut = (Expression<Timespan>) exprs[5 - matchedPattern];
+		stay = (Expression<Timespan>) exprs[3 - matchedPattern];
+		fadeIn = (Expression<Timespan>) exprs[4 - matchedPattern];
+		fadeOut = (Expression<Timespan>) exprs[5 - matchedPattern];
+		if (title != null && subtitle != null) {
+			return LiteralUtils.canInitSafely(title, subtitle);
+		} else if (title != null) {
+			return LiteralUtils.canInitSafely(title);
+		} else {
+			return LiteralUtils.canInitSafely(subtitle);
 		}
-		return true;
 	}
-	
+
 	@SuppressWarnings("null")
 	@Override
-	protected void execute(final Event e) {
-		String title = this.title != null ? this.title.getSingle(e) : null;
-		String subtitle = this.subtitle != null ? this.subtitle.getSingle(e) : null;
-		
-		if (TIME_SUPPORTED) {
-			int fadeIn, stay, fadeOut;
-			fadeIn = stay = fadeOut = -1;
+	protected void execute(final Event event) {
+		String title = this.title != null ? Classes.toString(this.title.getSingle(event)) : "";
+		String subtitle = this.subtitle != null ? Classes.toString(this.subtitle.getSingle(event)) : null;
+		int fadeIn, stay, fadeOut;
 
-			if (this.fadeIn != null) {
-				Timespan t = this.fadeIn.getSingle(e);
-				fadeIn = t != null ? (int) t.getAs(Timespan.TimePeriod.TICK) : -1;
-			}
+		fadeIn = getTicks(this.fadeIn, event);
+		stay = getTicks(this.stay, event);
+		fadeOut = getTicks(this.fadeOut, event);
```

Comment:
```suggestion
		fadeIn = getTicks(this.fadeIn, event, 10);
		stay = getTicks(this.stay, event, 70);
		fadeOut = getTicks(this.fadeOut, event, 20);
```
Bukkit says that if you provide `-1` it would use the last sent options that was passed through. We should be using the defaults for each one instead of `-1` if the user didnt provide the timespan themselves,

Reply:
Sovde told me thats what EffResetTitle is for, and base minecraft uses the same behavior as this.

Reply:
ahh ok, disregard this then